kimchi/src/circuits/gate.rs

```diff
@@ -68,40 +69,67 @@ pub struct SingleLookup<F> {
 /// analogously using `joint_combiner`.
 ///
 /// This function computes that combined value.
-pub fn combine_table_entry<'a, F: Field, I: DoubleEndedIterator<Item = &'a F>>(
+pub fn combine_table_entry<
+    'a,
+    F: 'a + Zero + One + Add + Mul + Clone,
```

Comment:
I'm wondering why `F: Field + Zero` here is not enough?

Also I'm wondering how I'm supposed to read `F: 'a`? And why it doesn't work without that

Reply:
`F` may be a type that is defined within a function; if the scope of `'a` is broader than the function, then the data outlives the type. Thus `F: 'a`.